src/main/groovy/org/pih/warehouse/inventory/CycleCountSubmitRecountCommand.groovy

```diff
@@ -9,9 +9,17 @@ class CycleCountSubmitRecountCommand extends CycleCountSubmitCountCommand {
 
     static constraints = {
         cycleCount(validator: { CycleCount obj ->
-            if (obj?.status != CycleCountStatus.INVESTIGATING) {
-                return ['invalidStatus']
-            }
+            // The validation should be uncommented after coming up with
+            // better solution for those cases:
+            // - enter qty > 0 while counting, then remove that inventory,
+            //   start recount (now cycle count has invalid status), we are
+            //   not able to finish the cycle count
+            // - remove inventory while recounting, reload, we are
+            //   not able to finish the cycle count because of
+            //   invalid cycle count status again
+            // if (obj?.status != CycleCountStatus.INVESTIGATING) {
+            //    return ['invalidStatus']
+            // }
```

Comment:
@EWaterman do you have any other better ideas how to make it work without removing the validation?
For now, there are two cases:
- enter qty > 0 while counting, then remove that inventory, start recount (now cycle count has invalid status - COUNTED), we are not able to finish the cycle count
- remove inventory while recounting, reload, we are not able to finish the cycle count because of invalid cycle count status again

Enabling creation items in that case with count index = 1 can destroy the current frontend behaviour

Reply:
The recent changes on the frontend that Kacper and I added stops you from submitting a recount if there are no count items. Because of that, I think these scenarios are now impossible to reach with the current UI. You'll want to test it out, but I think you can revert this change